build_utils/tests/intro.py

```diff
@@ -0,0 +1,223 @@
+###
+# Copy of intro notebook for the sake of testing new versions of the package.
```

Comment:
I would say let's move this to a pytest and run that test case instead of running the actual intro.py file. what do you think?

Reply:
Yes that's a good call, I'll do that. Shall I put it under tests/examples?

Reply:
I would say that would be a great place to put or we can move all of these tests to a new folder name `tests/integration` and keep them there, this way we can distinct it later to run them or not to run them where ever we want. 

I think it is good to separate these integration and maybe later we want to run them in their own CI step.

Reply:
I can help with setting up this one for you, it will be a little bit tricky, I will push it in another branch and add it to CI for you in couple days.

Reply:
https://github.com/stanfordnlp/dspy/pull/1048

I added my initial commits here.